go/arrow/avro/schema.go

```diff
@@ -0,0 +1,407 @@
+			sl := len(symbols)
+			switch {
+			case sl <= math.MaxUint8:
+				dt.IndexType = arrow.PrimitiveTypes.Uint8
+			case sl > math.MaxUint8 && sl <= math.MaxUint16:
+				dt.IndexType = arrow.PrimitiveTypes.Uint16
+			case sl > math.MaxUint16 && sl <= math.MaxUint32:
+				dt.IndexType = arrow.PrimitiveTypes.Uint32
+			}
+			c.arrowField = arrow.Field{Name: f.Name(), Type: &dt, Nullable: true, Metadata: arrow.MetadataFrom(symbols)}
+		// Avro "fixed" field type = Arrow FixedSize Primitive BinaryType
+		case *avro.FixedSchema:
+			schemaCache.Add(f.Name(), f.Type())
+			if isLogicalSchemaType(f.Type()) {
+				c := n.newChild(f.Name(), f.Type())
+				avroLogicalToArrowField(c)
+			} else {
+				c := n.newChild(f.Name(), f.Type())
+				arrowSchemafromAvro(c)
+			}
+		case *avro.RecordSchema:
+			schemaCache.Add(f.Name(), f.Type())
+			c := n.newChild(f.Name(), f.Type())
+			iterateFields(c)
+			// Avro "map" field type - KVP with value of one type - keys are strings
+		case *avro.MapSchema:
+			schemaCache.Add(f.Name(), ft.Values())
+			c := n.newChild(f.Name(), ft.Values())
+			arrowSchemafromAvro(c)
+			if ft.Values().Type() == "union" {
+				c.arrowField = arrow.Field{Name: f.Name(), Type: arrow.MapOf(arrow.BinaryTypes.String, arrow.BinaryTypes.String), Metadata: c.arrowField.Metadata, Nullable: true}
+			} else {
+				c.arrowField = arrow.Field{Name: f.Name(), Type: arrow.MapOf(arrow.BinaryTypes.String, c.arrowField.Type), Metadata: c.arrowField.Metadata, Nullable: true}
+			}
+		case *avro.UnionSchema:
+			if ft.Nullable() {
+				if len(ft.Types()) > 1 {
+					schemaCache.Add(f.Name(), ft.Types()[1])
+					c := n.newChild(f.Name(), ft.Types()[1])
+					c.union = true
+					arrowSchemafromAvro(c)
+				}
+			}
+		default:
+			schemaCache.Add(f.Name(), f.Type())
+			if isLogicalSchemaType(f.Type()) {
+				c := n.newChild(f.Name(), f.Type())
+				avroLogicalToArrowField(c)
+			} else {
+				c := n.newChild(f.Name(), f.Type())
+				arrowSchemafromAvro(c)
+			}
+
+		}
+	}
+	var fields []arrow.Field
+	for _, child := range n.Children() {
+		fields = append(fields, child.arrowField)
+	}
+
+	namedSchema, ok := isNamedSchema(n.schema)
+	if !ok || namedSchema == n.name+"_data" || !n.union {
+		n.arrowField = arrow.Field{Name: n.name, Type: arrow.StructOf(fields...), Nullable: true}
+	} else {
+		n.arrowField = arrow.Field{Name: n.name, Type: arrow.StructOf(fields...), Nullable: true, Metadata: arrow.MetadataFrom(map[string]string{"typeName": namedSchema})}
+	}
+
+}
+
+func isLogicalSchemaType(s avro.Schema) bool {
+	lts, ok := s.(avro.LogicalTypeSchema)
+	if !ok {
+		return false
+	}
+	if lt := lts.Logical(); lt != nil {
+		return true
+	}
+	return false
+}
+
+func isNamedSchema(s avro.Schema) (string, bool) {
+	if ns, ok := s.(avro.NamedSchema); ok {
+		return ns.FullName(), ok
+	}
+	return "", false
+}
+
+// Avro primitive type.
+//
+// NOTE: Arrow Binary type is used as a catchall to avoid potential data loss.
```

Comment:
instead of returning a "NotImplemented" error?

Reply:
default removed, no fallback needed.